File: android/app/src/main/java/com/phicdy/totoanticipation/model/storage/SettingStorageImpl.java
package com.phicdy.totoanticipation.model.storage;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;

public class SettingStorageImpl implements SettingStorage {

    private final SharedPreferences preferences;
    private static final String KEY_PREF = "keyPrefSetting";
    private static final String KEY_NOTIFY_DEADLINE = "keyNotifyDeadline";
    public SettingStorageImpl(@NonNull Context context) {
        preferences = context.getSharedPreferences(KEY_PREF, Context.MODE_PRIVATE);
    }

    @Override
    public boolean isDeadlineNotify() {
        return preferences.getBoolean(KEY_NOTIFY_DEADLINE, true);
    }

    @Override
    public void setDeadlineNotify(boolean isEnabled) {
        preferences.edit().putBoolean(KEY_NOTIFY_DEADLINE, isEnabled).apply();
    }
}

File: android/app/src/main/java/com/phicdy/totoanticipation/model/storage/SettingStorageImpl.kt
package com.phicdy.totoanticipation.model.storage

import android.content.Context

class SettingStorageImpl(context: Context) : SettingStorage {

    private val preferences = context.getSharedPreferences(KEY_PREF, Context.MODE_PRIVATE)

    override var isDeadlineNotify: Boolean
        get() = preferences.getBoolean(KEY_NOTIFY_DEADLINE, true)
        set(isEnabled) = preferences.edit().putBoolean(KEY_NOTIFY_DEADLINE, isEnabled).apply()

    override var isPrivacyPolicyAccepted: Boolean
        get() = preferences.getBoolean(KEY_PRIVACY_POLICY_ACCEPTED, false)
        set(isEnabled) = preferences.edit().putBoolean(KEY_PRIVACY_POLICY_ACCEPTED, isEnabled).apply()

    companion object {
        private const val KEY_PREF = "keyPrefSetting"
        private const val KEY_NOTIFY_DEADLINE = "keyNotifyDeadline"
        private const val KEY_PRIVACY_POLICY_ACCEPTED = "keyPrivacyPolicyAccepted"
    }
}
